WikiWatershed/model-my-watershed
# Add command for getting list of DRB users

## Overview

Adds a management command to get a list of all DRB users. For this, we took the [shapefile of ZIP codes from TigerLines](https://catalog.data.gov/dataset/tiger-line-shapefile-2015-2010-nation-u-s-2010-census-5-digit-zip-code-tabulation-area-zcta5-na) and intersected it with [the DRB perimeter](https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/mmw/settings/data/drb_perimeter.json) and got a [list of ZIP codes](https://github.com/WikiWatershed/model-my-watershed/files/1455542/zip2.csv.txt). See https://github.com/WikiWatershed/model-my-watershed/issues/2330#issuecomment-342974255

This list is then used to filter users that have listed such a ZIP code in their profile.

Connects #2330 

### Demo

```bash
> kj manage drbusers
+ ARGS=drbusers
+ [[ 1 -eq 1 ]]
+ [[ drbusers == runserver ]]
+ vagrant ssh app -c 'cd /opt/app && envdir /etc/mmw.d/env ./manage.py drbusers'
id,username,organization,user_type,country,postal_code
"1","test1","Azavea","University Professional or Research Staff","US","19103"
"8","test3","Azavea","Commercial/Professional","US","19103"
"12","test7","Azavea","Unspecified","US","19103"
Connection to 127.0.0.1 closed.
```

### Notes

Currently we have very little validation on user entry (for example we do not validate that ZIP codes are numeric and five digit, although I guess this to support international entry), so the data can be a little noisy.

For production, someone will have to SSH in to the App VM and then run:

```bash
envdir /etc/mmw.d/env python manage.py drbusers > drbusers.csv
```

## Testing Instructions

 * Check out this branch and run `./scripts/manage.sh drbusers`
 * Ensure all the correct users show up, and any users with missing ZIP codes or those outside the DRB are not included
